nomad/structs/structs.go

```diff
@@ -6118,22 +6133,34 @@ func (r *RecoverableError) IsRecoverable() bool {
 	return r.Recoverable
 }
 
-// Recoverable is an interface for errors to implement to indicate whether or
+func (r *RecoverableError) IsServerSideError() bool {
+	return r.ServerSideErr
+}
+
+// ErrorMeta is an interface for errors to implement to indicate whether or
 // not they are fatal or recoverable.
-type Recoverable interface {
+type ErrorMeta interface {
 	error
 	IsRecoverable() bool
+	IsServerSideError() bool
 }
 
 // IsRecoverable returns true if error is a RecoverableError with
 // Recoverable=true. Otherwise false is returned.
 func IsRecoverable(e error) bool {
-	if re, ok := e.(Recoverable); ok {
+	if re, ok := e.(ErrorMeta); ok {
```

Comment:
getter.GetError also implements `IsRecoverable` so we shouldn't change this assertion. Maybe keep the `Recoverable interface` as is and add:

```go
type ErrorMeta interface {
  Recoverable
  IsServerSideError
}
```

...or we could implement `IsServerSideError` on `getter.GetError` ... the meaning there seems a bit ambiguous though. Would a 400 fetching an artifact be a server-side error? I think _no_ because `Server` refers to a _Nomad Server_, but we'd still want to document the interface's exact behavior thoroughly.